### ISSUE TITLE
Spike - investigate CI data race flake

### ISSUE THREAD

**Issue by MitchAman** [solo-io/gloo#5860]
### Version

1.11.x (beta)

### Describe the solution you'd like

Seems like we've hit a data flake when creating shared informers in parallel (https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=bf7ed8b8-a624-4cdf-9c85-c405d7ebd610) and this has happened once or twice recently.
@Nathan Fudenberg :clap: identified this yesterday here: https://solo-io-corp.slack.com/archives/CFK0DQADU/p1643660236295319?thread_ts=1643654827.835399&cid=CFK0DQADU
and a path forward is mentioned here: https://github.com/kubernetes/kubernetes/pull/89019#issuecomment-600278461
Can we investigate what it takes to resolve this? If the work is non-trivial can we write up an issue outlining the bug/impact and we can dedicate eng resources to resolve in an upcoming sprint. If it is trivial then maybe we just try to knock it out

**Comment by sam-heilbron** [comment]
Same as 2 of the issues in https://github.com/solo-io/gloo/issues/5490

**Comment by sam-heilbron** [comment]
This has been merged into 1.12 -> 1.9.
